supertokens_python/utils.py

```diff
@@ -170,7 +170,10 @@ def utf_base64encode(s: str) -> str:
 
 
 def utf_base64decode(s: str) -> str:
-    return b64decode(s.encode("utf-8")).decode("utf-8")
+    # Adding extra "==" based on
+    # https://stackoverflow.com/questions/2941995/python-ignore-incorrect-padding-error-when-base64-decoding
+    # Otherwise it can raise "incorrect padding" error
+    return b64decode(s.encode("utf-8") + b"==").decode("utf-8")
```

Comment:
Please use a proper urlsafe b64 decode function if possible (`urlsafe_b64decode` ? )

Reply:
I tried but I was still getting "incorrect padding" error. Adding this extra b"==" solved it. Will check again.

Reply:
Confirmed. Just using `urlsafe_b64decode(s.encode("utf-8")).decode("utf-8")` doesn't solve it. I still get "incorrect padding" error.

Reply:
Try yourself: `'eyJraWQiOiJkLTE2ODI0OTY4MzcyMjUiLCJ0eXAiOiJKV1QiLCJ2ZXJzaW9uIjoiMyIsImFsZyI6IlJTMjU2In0'`



```
>> s = 'eyJraWQiOiJkLTE2ODI0OTY4MzcyMjUiLCJ0eXAiOiJKV1QiLCJ2ZXJzaW9uIjoiMyIsImFsZyI6IlJTMjU2In0'
>>> b64decode(s.encode("utf-8")).decode("utf-8")
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/usr/lib/python3.7/base64.py", line 87, in b64decode
    return binascii.a2b_base64(s)
binascii.Error: Incorrect padding
```

Reply:
hmm, b64 is handled a bit strangely in python I see.
Anyway, try to make sure this access token: 
```
eyJraWQiOiJkLTE2ODI1OTgyNTU3ODIiLCJ0eXAiOiJKV1QiLCJ2ZXJzaW9uIjoiMyIsImFsZyI6IlJTMjU2In0.eyJzdWIiOiLDv8O_w7_Dv8O_w7_Dv8O_w7_Dv8O_w7_Dv8O_w7_Dv8O_w7_Dv8O_w7_Dv8O_w7_DvyIsImV4cCI6MTY4MjYwMTg1NSwiaWF0IjoxNjgyNTk4MjU1LCJzZXNzaW9uSGFuZGxlIjoiOTMwMzAxNjItOTI4OC00ZjNlLWIxZWQtODEwY2U0NDU2NDI4IiwicmVmcmVzaFRva2VuSGFzaDEiOiJhNTdiYjgyY2RmODkzNTE4ZDc3MjJkNGY3MWFjZjljZmY1N2E1M2MyOWEzMjQwNmRiNzc0ZTI3NzllMjBjMjdmIiwicGFyZW50UmVmcmVzaFRva2VuSGFzaDEiOm51bGwsImFudGlDc3JmVG9rZW4iOm51bGwsImlzcyI6Imh0dHBzOi8vYXBpLnN1cGVydG9rZW5zLmlvL2F1dGgifQ.qYDv5BAaQSK71Mp1V88brbUSlw9f7_sDRHNs2WO5o_I0rJG7sQjJ4T_TUkrAt0PXAy30ILXMvY-U0MDI6puNBYECJVAGBjCLdask2uivonSpj9hLYmw0ZAnzy1S4xwwjmJr_i6tpn5J0yttjnhEUgPnCRwkStt8iGMfBJmRhfQZCxXDFiZzJrtpiTcrdgFLr1JPParinRF8niAI8vaRpSSQO2Jkx-UxlXmaXvWsURE0PF9tmJYb9caVjf1pf6kI1xmE2hob5g5bihgmLZWRyUmCpJMYtA1u8lmFzEsRNpnRKzRfTG1-PgGnaWbhjjmV6jbCVcqT27p8HotAZqk6OdQ
```
decodes right.

Reply:
```python
>>> b64decode(s.split(".")[0].encode("utf-8") + b"==")
b'{"kid":"d-1682598255782","typ":"JWT","version":"3","alg":"RS256"}'
>>> b64decode(s.split(".")[1].encode("utf-8") + b"==")
b'{"sub":"\xc3\xbf\xc3\xb0\xec;\xfc;\x0e\xc3\xbf\xc3\xb0\xec;\xfc;\x0e\xc3\xbf\xc3\xb0\xec;\xfc;\x0e\xc3\xbf\xc3\xb0\xec;\xfc;\x0e\xc3\xbf","exp":1682601855,"iat":1682598255,"sessionHandle":"93030162-9288-4f3e-b1ed-810ce4456428","refreshTokenHash1":"a57bb82cdf893518d7722d4f71acf9cff57a53c29a32406db774e2779e20c27f","parentRefreshTokenHash1":null,"antiCsrfToken":null,"iss":"https://api.supertokens.io/auth"}'
>>> b64decode(s.encode("utf-8") + b"==")
b'{"kid":"d-1682598255782","typ":"JWT","version":"3","alg":"RS256"}\x1e\xc8\x9c\xddX\x88\x8e\x88\xb0\xef\xf0\xec;\x0e\xff\x0e\xc3\xb0\xef\xf0\xec;\x0e\xff\x0e\xc3\xb0\xef\xf0\xec;\x0e\xff\x0e\xc3\xb0\xef\xf0\xec;\x0e\xff\x0e\xc3\xb0\xef\xc8\x8b\x08\x99^\x1c\x08\x8e\x8cM\x8e\x0c\x8d\x8c\x0cN\rMK\x08\x9aX]\x08\x8e\x8cM\x8e\x0c\x8dNN\x0c\x8dMK\x08\x9c\xd9\\\xdc\xda[\xdb\x92\x18[\x99\x1b\x19H\x8e\x88\x8eL\xcc\x0c\xcc\x0cM\x8c\x8bNL\x8e\x0e\x0bM\x19\x8c\xd9KX\x8cYY\x0bN\x0cL\x18\xd9M\r\rM\x8d\x0c\x8e\x08\x8b\x08\x9c\x99Y\x9c\x99\\\xda\x15\x1b\xda\xd9[\x92\x18\\\xda\x0cH\x8e\x88\x98MM\xd8\x98\x8e\x0c\x98\xd9\x19\x8e\x0eL\xcdLN\x19\r\xcd\xcc\x8c\x99\r\x19\x8d\xccXX\xd9\x8eX\xd9\x99\x8dM\xd8ML\xd8\xcc\x8eXL\xcc\x8d\x0c\r\x99\x18\x8d\xcd\xcd\x19L\x8d\xcd\xceYL\x8c\x18\xcc\x8d\xd9\x88\x8b\x08\x9c\x18\\\x99[\x9d\x14\x99Y\x9c\x99\\\xda\x15\x1b\xda\xd9[\x92\x18\\\xda\x0cH\x8e\x9b\x9d[\x1b\x0b\x08\x98[\x9d\x1aP\xdc\xdc\x99\x95\x1b\xda\xd9[\x88\x8e\x9b\x9d[\x1b\x0b\x08\x9a\\\xdc\xc8\x8e\x88\x9a\x1d\x1d\x1c\x1c\xce\x8b\xcb\xd8\\\x1aK\x9c\xdd\\\x19\\\x9d\x1b\xda\xd9[\x9c\xcb\x9a[\xcb\xd8]]\x1a\x08\x9fB\xa6\x03\xbf\x90@i\x04\x8a\xefS)\xd5_<n\xb6\xd4J\\=\x7f\xbb\x03Dsl\xd9c\xb9\xa0\x8d+$n\xecB2xM5$\xac\x0bt=p2\xdfB\x0b\\\xcb\xd8SC\x03#\xaan4\x16\x04\x08\x95@\x18\x18\xc2-\xd6\xac\x93k\xa2\xbe\x89\xd2\xa6?a-\x89\xb0\xd1\x90\'\xcf-R\xe3\x1c0\x8ebk\x8b\xabi\x9f\x92t\xca\xdbc\x9e\x11\x14\x80\xf9\xc2G\t\x12\xb6\xdf"\x18\xc7\xc1&da}\x06B\xc5p\xc5\x89\x9c\xc9\xae\xdabM\xca\xdd\x80R\xeb\xd4\x93\xcfj\xb8\xa7D_\'\x88\x02<\xbd\xa4iI$\x0e\xd8\x991S\x19W\x99\xa5\xefZ\xc5\x11\x13C\xc5\xf6\xd9\x89a\xbf\\iX\xdf\xd6\x97\xfa\x90\x8dq\x98M\xa1\xa1\xbe`\xe5\xb8\xa1\x82b\xd9Y\x1c\x94\x98*I1\x8b@\xd6\xef%\x98\\\xc4\xb1\x13i\x9d\x12\xb3E\xf4\xc6\xd4\xf8\x06\x9d\xa5\x9b\x868\xe6W\xa8\xdb\tW*On\xe9\xf0z-\x01\x9a\xa4\xe8\xe7P'
```

Reply:
Okay turns out we need both. Using this now:
```python
urlsafe_b64encode(s.encode("utf-8")).decode("utf-8")
```

Reply:
Done.